bson/src/test/scala/org/mongodb/scala/bson/codecs/MacrosSpec.scala

```diff
@@ -141,6 +145,11 @@ class MacrosSpec extends FlatSpec with Matchers {
     decode(registry.get(classOf[OptionalValue]), buffer) should equal(OptionalValue("Bob", None))
   }
 
+  it should "be able to round trip ignored optional values" in {
+    roundTrip(OptionalValueIgnoreNone("Bob", None), """{name: "Bob"}""", classOf[OptionalValueIgnoreNone])
```

Comment:
Though Option<T> is preferred to null in Scala, the behavior of actual null values should be defined.  Currently both of these tests result in an NPE:

    roundTrip(OptionalValueIgnoreNone("Bob", null), """{"name": "Bob", value: null}""", classOf[OptionalValueIgnoreNone]);
    roundTrip(OptionalValueIgnoreNone(null, None), """{"name": null}""", classOf[OptionalValueIgnoreNone]);